networking/external_dns_operator/nw-control-dns-records-public-managed-zone-gcp.adoc

```diff
@@ -0,0 +1,148 @@
++
+[source,terminal]
+----
+$ export GOOGLE_CREDENTIALS=decoded-gcloud.json
+----
+
+. Activate your account by using the following command:
++
+[source,terminal]
+----
+$ gcloud auth activate-service-account  <client_email as per decoded-gcloud.json> --key-file=decoded-gcloud.json
+----
+
+. Set your project by running the following command:
++
+[source,terminal]
+----
+$ gcloud config set project <project_id as per decoded-gcloud.json>
+----
+
+. Get the routes to check the domain by running the following command:
++
+[source,terminal]
+----
+$ oc get routes --all-namespaces | grep console
+----
++
+.Example output
+[source,terminal]
+----
+openshift-console          console             console-openshift-console.apps.qe.gcp.devcluster.openshift.com                       console             https   reencrypt/Redirect     None
+openshift-console          downloads           downloads-openshift-console.apps.qe.gcp.devcluster.openshift.com                     downloads           http    edge/Redirect          None
+----
+
+. Get your zone that installer created for you. Run the following command:
++
+[source,terminal]
+----
+$ gcloud dns managed-zones list | grep qe.gcp.devcluster.openshift.com
+qe-cvs4g-private-zone qe.gcp.devcluster.openshift.com
+----
+
+. Check if the zone only have DNS records either `NS` or `SOA`:
++
+[source,terminal]
+----
+$ gcloud dns record-sets list --zone=qe-cvs4g-private-zone
+----
+
+. Create a `ExternalDNS` CR for `route`. Run the following command:
++
+[source,yaml]
+----
+apiVersion: externaldns.olm.openshift.io/v1alpha1
+kind: ExternalDNS
+metadata:
+  name: sample-gcp <1>
+spec:
+  domains:
+    - filterType: Include <2>
+      matchType: Exact <3>
+      name: qe.gcp.devcluster.openshift.com <4>
+  provider:
+    type: GCP <5>
+  source:
+    openshiftRouteOptions: <6>
+      routerName: default <7>
+    type: OpenShiftRoute <8>
+----
+
+<1> The `name` defines the name of External DNS CR.
+<2> By default all hosted zones are selected as potential targets. You can include a hosted zone that you need.
+<3> The matching of the target zone's domain has to be exact (as opposed to regular expression match).
+<4> Specify the exact domain of the zone you want to update. The hostname of the routes must be subdomains of the specified domain.
+<5> The `type` defines Google Cloud DNS provider.
+<6> You can define an option for route source under `openshiftRouteOptions` field.
+<7> If the source is `openshift-route` then you can pass the ingress controller name. Based on this name external-dns will select the respective router from the route status and map that routerCanonicalHostname to the route host while creating a CNAME record.
+<8> Defines source route, dns records for routes get created on your cloud DNS.
+
+. Check the records created for OCP routes using the following command:
++
+[source,terminal]
+----
+$ gcloud dns record-sets list --zone=qe-cvs4g-private-zone
```

Comment:
```suggestion
$ gcloud dns record-sets list --zone=qe-cvs4g-private-zone | grep console
```
Just as for Azure, let's use the console routes, this will simplify the doc.

Reply:
Addressed